src/main/java/com/google/sps/data/DatastoreManager.java

```diff
@@ -0,0 +1,192 @@
+        postEntity.setProperty(Constants.POST_ANSWER, -1); 
+        postEntity.setProperty(Constants.POST_REPLIES, new ArrayList<>()); 
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(postEntity); 
+        return postEntity.getKey().getId(); 
+    }
+
+    /**
+     * Updates a representative entity with a new post entity 
+     * @param repId ID of the representative entity 
+     * @param postId ID of the post entity 
+     */ 
+    public static void updateRepresentativePostList(long repId, long postId) throws EntityNotFoundException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Key repEntityKey = KeyFactory.createKey(Constants.REP_ENTITY_TYPE, repId);
+        Entity repEntity = (Entity) datastore.get(repEntityKey); 
+        List<Long> postIds = (ArrayList<Long>) repEntity.getProperty(Constants.REP_POSTS); 
+        if (postIds == null) {
+            postIds = new ArrayList<>(); 
+        }
+        postIds.add(postId);
+        repEntity.setProperty(Constants.REP_POSTS, postIds); 
+        datastore.put(repEntity);
+    }
+
+    /**
+     * Updates a post entity with a comment in the replies list 
+     * @param postId ID of the post entity 
+     * @param commentId ID of the comment being added as a reply 
+     */ 
+    public static void updatePostWithComment(long postId, long commentId) throws EntityNotFoundException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity postEntity = DatastoreManager.queryForPostEntityWithId(postId); 
+        List<Long> commentIds = (ArrayList<Long>) postEntity.getProperty(Constants.POST_REPLIES); 
+        if (commentIds == null) {
+            commentIds = new ArrayList<>(); 
+        }
+        commentIds.add(commentId); 
+        postEntity.setProperty(Constants.POST_REPLIES, commentIds); 
+        datastore.put(postEntity);
+    }
+
+    /**
+     * Updates a post entity with a comment in the replies list 
+     * @param postId ID of the post entity 
+     * @param commentId ID of the comment being added as a reply 
+     */ 
+    public static void updatePostWithAnswer(long postId, long answerId) throws EntityNotFoundException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity postEntity = DatastoreManager.queryForPostEntityWithId(postId); 
+        postEntity.setProperty(Constants.POST_ANSWER, answerId); 
+        datastore.put(postEntity);
+    }
+
+    /**
+     * Searches datastore for a particular representative and converts the entity 
+     * to a Representative instance 
+     * @param repName name of the representative to search datastore for 
+     * @return the representative object, or null if it was not found in datastore 
+     */ 
+    public static @Nullable Representative queryForRepresentativeObjectWithName(String repName) {
```

Comment:
you aren't consistent with the use of Nullable. While I prefer it, i'm ok if you get rid of it. just be consistent otherwise it isn't useful.